### PR TITLE
add the function of restoring from metadata been backuped

### PR DESCRIPTION
**Pull request by yangyuliufeng**
### Ⅰ. Describe what this PR does
This PR edits the crd of dataset, so that user can define a restore location to restore from metadata been backuped
It also change the logic of loading metadata and getting ufs total, fileNum
### Ⅱ. Does this pull request fix one issue?

#384

### Ⅳ. Describe how to verify it
@XIAO-HOU has verfied it.

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Review by xieydd — CHANGES_REQUESTED** [review]
how about fix the ci check

File: pkg/ddc/alluxio/transform.go [hunk 1]
```diff
@@ -256,7 +257,33 @@ func (e *AlluxioEngine) transformMasters(runtime *datav1alpha1.AlluxioRuntime, v
 	// 	value.Master.Env["ALLUXIO_UID"] = strconv.FormatInt(*runtime.Spec.RunAs.UID, 10)
 	// 	value.Master.Env["ALLUXIO_GID"] = strconv.FormatInt(*runtime.Spec.RunAs.GID, 10)
 	// }
-
+	// if the dataset indicates a restore path, need to load the  backup file in it
+	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
+	if err == nil {
```
Comment (cheyang) [inline pkg/ddc/alluxio/transform.go:262]: `if err != nil`是一个比较直接的逻辑。另外建议增加一下注释

File: pkg/ddc/alluxio/operations/base.go [hunk 2]
```diff
@@ -191,6 +191,36 @@ func (a AlluxioFileUtils) InsertMetaDataInfoIntoFile(key KeyOfMetaDataFile, valu
 	return err
 }
 
+// QueryMetadataInfoFile query the metadata info file.
+func (a AlluxioFileUtils) QueryMetaDataInfoIntoFile(key KeyOfMetaDataFile, filename string) (value string, err error) {
+	line := ""
+	switch key {
+	case DatasetName:
```
Comment (cheyang) [inline pkg/ddc/alluxio/operations/base.go:198]: 能否增加文件内容的例子，方便理解这段代码的逻辑。

File: api/v1alpha1/dataset_types.go [hunk 3]
```diff
@@ -129,6 +139,10 @@ type DatasetSpec struct {
 	// +kubebuilder:validation:Enum=Exclusive;"";Shared
 	// +optional
 	PlacementMode PlacementMode `json:"placement,omitempty"`
+
+	// DataRestoreLocation is the location to load data of dataset  been backuped
+	// +optional
+	DataRestoreLocation DataRestoreLocation `json:"dataRestoreLocation,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/dataset_types.go:145]: 可否用指针？

File: pkg/ddc/alluxio/metadata.go [hunk 4]
```diff
@@ -152,48 +161,91 @@ func (e *AlluxioEngine) syncMetadataInternal() (err error) {
...
-			// get total size
-			datasetUFSTotalBytes, err := e.TotalStorageBytes()
-			if err != nil {
-				e.Log.Error(err, "Get Ufs Total size failed when syncing metadata", "name", e.name, "namespace", e.namespace)
-				result.Err = err
-				result.Done = false
-				resultChan <- result
-				return
+			if result.Done == false {
+				result.Err = errors.New("GetMetadataInfoFailed")
```
Comment (TrafalgarZZZ) [inline pkg/ddc/alluxio/metadata.go:245]: 这里是不是根据具体的错误结果返回更好? 可以更精准地得知哪一步出了问题.

File: docs/zh/samples/backup_and_restore_metadata.md [hunk 5]
```diff
@@ -114,4 +114,101 @@ $ kubectl describe node cn-beijing.192.168.1.146
...
+数据集加载时，将不再从UFS中加载metadata并统计UFS TOTAL SIZE等信息，而是从备份文件中恢复
+
+片刻后，Dataset进入Bound状态：
+```bash
+$ kubectl get dataset
+NAME    UFS TOTAL SIZE   CACHED   CACHE CAPACITY   CACHED  PERCENTAGE   PHASE   AGE
+hbase   443.86MiB        0.00B    4.00GiB          0.0%                 Bound   20h
+```
+### 从本地恢复
+在创建Datast时，在spec中添加dataRestoreLocation，填入刚刚查询到的路径和备份文件所在主机的nodeName
```
Comment (TrafalgarZZZ) [inline docs/zh/samples/backup_and_restore_metadata.md:186]:
```suggestion
在创建Dataset时，在spec中添加dataRestoreLocation，填入刚刚查询到的路径和备份文件所在主机的nodeName
```

File: docs/zh/samples/backup_and_restore_metadata.md [hunk 6]
```diff
@@ -114,4 +114,101 @@ $ kubectl describe node cn-beijing.192.168.1.146
 获得IP地址后，就可以从主机的对应目录取走文件刚刚的两个文件
 
 
-## 恢复
\ No newline at end of file
+## 恢复
+要进行恢复，需要保证Dataset的名称与原来保持一致，否则会找不到备份文件
+
+### 从PVC恢复
+在创建Datast时，在spec中添加dataRestoreLocation，填入刚刚查询到的路径作为恢复路径
```
Comment (TrafalgarZZZ) [inline docs/zh/samples/backup_and_restore_metadata.md:121]:
```suggestion
在创建Dataset时，在spec中添加dataRestoreLocation，填入刚刚查询到的路径作为恢复路径
```

File: pkg/ddc/alluxio/metadata.go (hunk 4 above)
Comment (yangyuliufeng) [reply pkg/ddc/alluxio/metadata.go:245]: > 这里是不是根据具体的错误结果返回更好? 可以更精准地得知哪一步出了问题.

这里有可能发生了多种错误，所以我统一回了一个GetMetadataInfoFailed

Comment (cheyang) [reply pkg/ddc/alluxio/metadata.go:245]: 具体错误如何诊断？有什么办法不？

**Pull request merged**